core/common/src/ThematicDisplay.ts

```diff
@@ -119,6 +119,43 @@ export class ThematicGradientSettings {
     return true;
   }
 
+  /** Compares two sets of thematic gradient settings.
+   * @param lhs First set of thematic gradient settings to compare
+   * @param rhs Second set of thematic gradient settings to compare
+   * @returns 0 if lhs is equivalent to rhs, a negative number if lhs compares less than rhs, or a positive number if lhs compares greater than rhs.
+   */
+  public static compare(lhs: ThematicGradientSettings, rhs: ThematicGradientSettings): number {
+    if (lhs.mode < rhs.mode)
```

Comment:
Simplify.
```
let diff = 0;
if (diff = compareNumbers(lhs.mode, rhs.mode))
  return diff;
else if (diff = compareNumbers(lhs.stepCount, rhs.stepCount))
  return diff;
else if (diff = compareNumbers(lhs.marginColor.tbgr, rhs.marginColor.tbgr))
  return diff;
// etc...
```

Reply:
Handy function.  Need to use compareWithTolerance(), though, am I right?  (Need to return -1, 0, or 1)

Reply:
Oh, we just expect negative, positive or zero.  Will work out fine.  Need to change my test - I was expecting something different.  Sounds good.

Reply:
Simplified it up.